Allow smtps:// in MAIL_URL to enable secureConnection

MichaelJCole opened the pull request:
This backwards compatible patch allows smtps protocol, and secureConnection on port 587.  Port 465 is non-standard and assigned to something else.  See #7042 for details.

MichaelJCole commented:
Please see #7042

abernix commented on packages/email/email.js:
```diff
@@ -17,9 +17,9 @@ var MailComposer = EmailInternals.NpmModules.mailcomposer.module.MailComposer;
+  if (mailUrl.protocol !== 'smtp:' && mailUrl.protocol !== 'smtps:')
     throw new Error("Email protocol in $MAIL_URL (" +
-                    mailUrlString + ") must be 'smtp'");
+                    mailUrlString + ") must be 'smtp' or 'smtps'");
```
This only changes whether or not an error is thrown when `smtps` is used.  It doesn't change `secureConnection` below.

abernix commented:
This doesn't appear to change the `secureConnection` value, something must be missing.

abernix commented:
You had this code in the Issue
```
{ secureConnection: (port === 465) || (mailUrl.protocol === 'smtps:'),
```
...but it's not here in the PR.  This makes the difference between this change working and not working by allowing the additional trigger for `secureConnection`, which would definitely be necessary if the provider didn't support 465.

Pretty strange for SparkPost to just not support it though. I'm aware it is not RFC and it's silly, but it is widely accepted by many (Gmail, Hotmail, Mandrill, SendGrid, MailGun, AWS SES, etc.).  Looks like Postmarkapp doesn't support it either.

Want to update this PR?

MichaelJCole commented:
@abernix Hey Jesse, there are steps in git, and I missed one.  Added the 2nd commit. Good catch.

Yeah, this PR didn't fix the SparkPost issue.  I'm in a hurry and switched to Mailgun instead.  the 'simpleSMTP' npm package is depricated.  Next steps for Sparkpost issue would be to switch to a new mail transport library, but that's a bigger choice than a drive-by PR :-)

carinakoo commented:
@MichaelJCole, thanks for this leg work. I'm experiencing this as well with Sparkpost. Do I understand correctly then that there's no known fix for Sparkpost? Enabling `secureConnection` did not solve? Thanks.

MichaelJCole commented:
Hey @carinakoo I was in a hurry and switched to mailgun with no problems yet.  You might get a better response from Sparkpost because you're not the first.  Give their support a try.

zol commented:
Looks great, thanks everyone!
